web/TelephoneLink.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace luya\web;
+
+use luya\cms\models\NavItemPageBlockItem;
+use luya\tag\tags\TelTag;
+use yii\base\BaseObject;
+use yii\validators\RegularExpressionValidator;
+
+/**
+ * Telephone Link.
+ *
+ * Represent a {{luya\web\LinkInterface}} of an telephone link.
+ *
+ * @author Bennet Klarhölter <boehsermoe@me.com>
+ * @since 1.0.9
+ */
+class TelephoneLink extends BaseObject implements LinkInterface
+{
+    use LinkTrait;
+
+    private $_telephone;
+
+    /**
+     * Setter method for telephone number.
+     *
+     * If no valid telephone is provided, not value is set.
+     *
+     * @param string $telephone The telephone number which should be used for the tel link.
+     */
+    public function setTelephone($telephone)
+    {
+        /**
+         * Hack to support leading + sign
+         * @see ArrayHelper::typeCast in NavItemPageBlockItem::rules()
```

Comment:
I don't understand this information, could you please explain? Sorry

Reply:
When save a number like "+123456" in the `LinkButtonBlock` this string value will be cast to an integer by `ArrayHelper::typeCast` in [\luya\cms\models\NavItemPageBlockItem::rules()](https://github.com/luyadev/luya-module-cms/blob/abd5cd8490fa52be2df0ab5548ee652e79c427a3/src/models/NavItemPageBlockItem.php#L74) and remove the "+" sign. With "\+123456" it will not be cast to an integer.
So I implemented a hack with removing leading "\". Maybe you have a idea how to solve this problem?

Reply:
If the ltrim has no effect, assuming this test case:

```php
$this->assertSame([123], ArrayHelper::typeCast([ltrim('+123', '\\')]));
```